apps/api/src/app/sessions/sessions.service.ts

```diff
@@ -46,34 +90,29 @@ export class SessionsService {
     });
   }
 
-  async update(id: number, updateSessionInput: UpdateSessionInput): Promise<Session> {
-    try {
-      await this.sessionsRepository.update(id, {
-        name: updateSessionInput.name,
-        description: updateSessionInput.description,
-        status: updateSessionInput.status,
-        start: updateSessionInput.start,
-        end: updateSessionInput.end,
-        editable: updateSessionInput.editable,
-        enablePlayer: updateSessionInput.enablePlayer,
-        displaySampleSolution: updateSessionInput.displaySampleSolution,
-        enableLiveAnalysis: updateSessionInput.enableLiveAnalysis,
-        ...(updateSessionInput.mediaId && { media: { id: updateSessionInput.mediaId } }),
-      });
-
-      return this.findOne(id);
-    } catch (error) {
-      throw new BadRequestException(error.detail);
-    }
-  }
+  async update(id: number, updateSessionInput: UpdateSessionInput, owner: User): Promise<Session> {
+    const session = await this.findOneOfOwner(id, owner);
 
-  async remove(id: number) {
-    const deleteResult = await this.sessionsRepository.softDelete(id);
+    this.sessionsRepository.merge(session, {
```

Comment:
Theoretically the session could be undefined/null, because it is not guaranteed before that the is belongs to the user as owner right?

Reply:
When it is null an NotFoundException is thrown by findONeOfOwner